src/crimson/osd/recovery_backend.h

```diff
@@ -50,6 +50,18 @@ class RecoveryBackend {
     assert(it->second);
     return {*(it->second), added};
   }
+  std::pair<seastar::future<>, bool> add_unfound(const hobject_t &soid) {
+    auto [it, added] = unfound.emplace(soid, seastar::shared_promise());
+    return {it->second.get_shared_future(), added};
```

Comment:
We can return `seastar::future<> add_unfound` instead of the std::pair since there are no users to the returned boolean.